src/main/java/net/snowflake/ingest/streaming/internal/RegisterService.java

```diff
@@ -140,14 +140,28 @@ List<FlushService.BlobData> registerBlobs(Map<String, Timer.Context> latencyTime
                 retry++;
                 break;
               }
-              logger.logError(
-                  "Building or uploading blob={} failed or timed out, exception={}, detail={},"
-                      + " cause={}, detail={}, all channels in the blob will be invalidated",
-                  futureBlob.getKey().getFilePath(),
-                  e,
-                  e.getMessage(),
-                  e.getCause(),
-                  e.getCause() == null ? null : e.getCause().getMessage());
+              StringBuilder stackTrace = new StringBuilder();
+              if (e.getCause() != null) {
+                for (StackTraceElement element : e.getCause().getStackTrace()) {
+                  stackTrace.append(System.lineSeparator()).append(element.toString());
+                }
+              }
```

Comment:
I think there might be a helper somewhere to do this in a common Java utility library but this is fine.